builders/tslice.py

```diff
@@ -0,0 +1,190 @@
+                                      "dim": "time",
+                                      "coords": "minimal",
+                                      "compat": "override"
+                                      }
+                           },
+           ] 
+        }                         
+    }
+    # add a section for each column
+    for col in cols:
+        c = {
+            "column_name": col,
+            "vocabulary": ""
+            }
+        d["attributes"].append(c)
+
+    od=OrderedDict(d.items())
+    #write the json file
+    with open(json_filepath,'w') as f:
+        json.dump(od,f,indent=4)
+
+
+def verify(input_yaml):
+    # verify that we're working with a dictionary
+    if not isinstance(input_yaml,dict):
+        print("ERROR: The experiment/dataset top level is not a dictionary. Make sure you follow the correct format.")
+        return False
+    for dataset in input_yaml.keys():
+        # check to see if there is a data_sources key for each dataset
+        if 'data_sources' not in input_yaml[dataset].keys():
+            print("ERROR: Each experiment/dataset must have the key 'data_sources'. Verify "+dataset+" contains this key.")
+            return False
+        # verify that we're working with a list at this level
+        if not isinstance(input_yaml[dataset]['data_sources'],list):
+            print("ERROR: The data_sources are not in a list.  Make sure you follow the correct format.")
+            return False
+        for stream_info in input_yaml[dataset]['data_sources']:
+            # check to make sure that there's a 'glob_string' key for each data_source
+            if 'glob_string' not in stream_info.keys():
+                print("ERROR: Each data_source must contain a 'glob_string' key.  Verify that all data_sources under "+dataset+" contain a 'glob_string' key.")
+                return False
+    return True
+
+
+def common_parser(filepath, local_attrs, glob_attrs):
+    """
+    gather the file parts for the df
+    """
+    basename = os.path.basename(filepath)
+    fileparts = {}
+    fileparts['path'] = filepath
+
+    try:
+        fileparts['variable'] = []
+        # open file
+        d = nc.Dataset(filepath,'r')
+        # find what the time (unlimited) dimension is
+        dims = list(dict(d.dimensions).keys())
+        look_for_unlim=[d.dimensions[dim].isunlimited() for dim in dims]
+        unlim=[i for i, x in enumerate(look_for_unlim) if x]
+        unlim_dim=dims[unlim[0]]
+        fileparts['time_range'] = str(d[unlim_dim][0])+'-'+str(d[unlim_dim][-1])
+        # loop through all variables
+        for v in d.variables.keys():
+            # test to see if this is a time varying field, if so, add to the catalog
+            if unlim_dim in d.variables[v].dimensions:
+                fileparts['variable'].append(v)
+        # add the keys that are common to all files in the dataset
+        for gv in glob_attrs.keys():
+            fileparts[gv] = glob_attrs[gv]
```

Comment:
Wouldn't it be better to let the user tells us what attributes to extract from the global attributes instead of extracting everything? As I mentioned in the past, this is likely going add so much noise to the catalog. For example, here's one WRF output file. It has **102** keys in the global attributes. This would result in a dataframe with 100+ columns. 


<img width="978" alt="Screen Shot 2020-04-21 at 12 50 35 PM" src="https://user-images.githubusercontent.com/13301940/79902449-be740e80-83ce-11ea-81aa-e2f09ee84a2d.png">

Reply:
This should be just looking at the user input and not the global attributes on the file.  So in the first group:
```
my_experiment1:
    experiment_name: cesm_experiment_1
    member_id: '001'
    data_sources:
    - glob_string: /the/path/to/data/CESM_DATA/cesm_experiment_1/atm/hist/*.cam.h0.*
      model_name: cam
      time_freq: month_1
    - glob_string: /the/path/to/data/CESM_DATA/cesm_experiment_1/lnd/hist/*clm2.h0.*
      model_name: clm
      time_freq: month_1
    - glob_string: /the/path/to/data/CESM_DATA/cesm_experiment_1/ocn/hist/*pop.h.*
      model_name: pop
      time_freq: month_1
    - glob_string: /the/path/to/data/CESM_DATA/cesm_experiment_1//ice/hist/*cice.h.*
      model_name: cice
      time_freq: month_1
```
The `glob_attrs` var should contain these key/value pairs:
```
experiment_name: cesm_experiment_1
member_id: '001'
data_sources:[....]
```
I named it global because all of the `data_sources` will have these same values under those columns.  The only things that it gets from the netcdf files that are put into the dataframe are the variable names.  Sorry for the confusion in terminology.

Reply:
> This should be just looking at the user input and not the global attributes on the file.


Thank you for the clarification! Would it still be useful to harvest some of global attributes present in the file? Or did we conclude that these global attributes from the file aren't useful? :)

Reply:
I don't think they're useful without a filter.  I can add this in, though, if someone thinks it would be useful.

Reply:
Let's omit them for the time being. I just looked at one of the CESM-CIMP6 files, and I was disappointed to find out that there's no useful information from the global attributes:

```python
{'Conventions': 'CF-1.0',
 'source': 'CAM',
 'case': 'b.e21.B1850.f19_g17.CMIP6-piControl-2deg.001',
 'logname': 'cmip6',
 'host': 'cheyenne3',
 'initial_file': 'b.e20.B1850.f19_g17.release_cesm2_1_0.020.cam.i.0301-01-01-00000.nc',
 'topography_file': '/glade/p/cesmdata/cseg/inputdata/atm/cam/topo/fv_1.9x2.5_nc3000_Nsw084_Nrs016_Co120_Fi001_ZR_GRNL_031819.nc',
 'model_doi_url': 'https://doi.org/10.5065/D67H1H0V',
 'time_period_freq': 'month_1'}
```